altizon/io/datonis/edge/edge_gateway_mqtt.py

```diff
@@ -16,13 +17,33 @@
     import Queue as Queue
     import thread as thread
 
-def on_connect(client, userdata, flags,rc):
-    logging.info("Connected to the MQTT broker with return code: " + str(flags) + ", " + str(rc))
-    userdata.subscribe_for_instructions()
-    userdata.subscribe_for_acks()
+CONNECTING = 0
+CONNECTED = 1
+DISCONNECTED = 2
+RECONNECTING = 3
+UNAUTHORISED = 4
+
+ALIOT_PROTOCOL_VERSION = 2.0
+
+def on_connect(client, userdata, flags, rc):
+    if rc == mqtt.CONNACK_ACCEPTED:
+        userdata.state = CONNECTED
+        logging.info("Connected to the MQTT broker with return code: " + str(flags) + ", " + str(rc))
+        userdata.subscribe_for_instructions()
+        userdata.subscribe_for_acks()
+    elif rc == mqtt.CONNACK_REFUSED_NOT_AUTHORIZED:
+        logging.error("Connection Unauthorised. ")
+        userdata.state = UNAUTHORISED
+        client.disconnect()
```

Comment:
Do you still need to explicitly disconnect? Or will disconnect raise an exception indicating already not connected?